plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -458,8 +459,13 @@ private DeltaLakeTransactionLogEntry buildAddEntry(ConnectorSession session, Blo
         log.debug("Block %s has %s fields", block, addEntryRow.getFieldCount());
         CheckpointFieldReader add = new CheckpointFieldReader(session, addEntryRow, type);
 
-        String path = add.getString("path");
         Map<String, String> partitionValues = add.getMap(stringMap, "partitionValues");
+        Map<String, Optional<String>> canonicalPartitionValues = canonicalizePartitionValues(partitionValues);
+        if (!(partitionConstraint.isAll() || partitionMatchesPredicate(canonicalPartitionValues, partitionConstraint.getDomains().orElseThrow()))) {
+            return null;
```

Comment:
Won't `buildTxnEntry` still be called before we bail out here ? If possible, we should try to avoid that.